config/kube.py

```diff
@@ -0,0 +1,359 @@
+
+c)  In Master check if all system pods are running good.
+    'kubectl get pods --namespace kube-system'
+
+    If any not running properly, say READY column has 0/1, decribe pod for more detail.
+    'kubectl --namespace kube-system describe pod <pod name>'
+
+    For additional details, look into pod's logs.
+    @ node: /var/log/pods/<podname>/...
+    @ master: 'kubectl logs -n kube-system <pod name>'
+    """
+
+    (h, fname) = tempfile.mkstemp(suffix="kube_hints_")
+    os.write(h, msg)
+    os.close(h)
+
+    _log_err("Refer file {} for troubleshooting tips".format(fname))
+    
+def _do_join(server,insecure):
+
+    try:
+        _download_file(server, insecure)
+
+        run_command("systemctl enable kubelet")
+
+        run_command(KUBEADM_JOIN_CMD.format(hostname()), ignore_error=True)
+
+        if _is_connected(server):
+            labels = _get_labels()
+            for label in labels:
+                _label_node(label)
+
+
+    except requests.exceptions.RequestException as e:
+        _do_exit("Download failed: {}".format(str(e)))
+
+    except OSError as e:
+        _do_exit("Download failed: {}".format(str(e)))
+
+    _troubleshoot_tips()
+
+
+def kube_reset():
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log_err("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    # Remove a key label and drain/delete self from cluster
+    # If not, the next join would fail
+    #
+    if os.path.exists(KUBE_ADMIN_CONF):
+        _label_node("enable_pods-")
+        run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s drain {} --ignore-daemonsets".format(
+                    KUBE_ADMIN_CONF, hostname()),
+                ignore_error=True)
+        run_command(
+                "kubectl --kubeconfig {} --request-timeout 20s delete node {}".format(
+                    KUBE_ADMIN_CONF, hostname()),
+                ignore_error=True)
+
+    run_command("kubeadm reset -f", ignore_error=True)
+    run_command("rm -rf /etc/cni/net.d")
+    run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    run_command("systemctl stop kubelet")
+    run_command("systemctl disable kubelet")
+
+def kube_join(force=False):
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log_err("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    db_data = _get_db_data('KUBERNETES_MASTER', 'SERVER')
+    if not db_data or 'IP' not in db_data or not db_data['IP']:
+        _log_err("Kubernetes server is not configured")
+
+    if db_data['disable'].lower() != "false":
+        _log_err("kube join skipped as kubernetes server is marked disabled")
+        return
+
+    if not force:
+        if _is_connected(db_data['IP']):
+            # Already connected. No-Op
+            return
+
+    kube_reset()
+    _do_join(db_data['IP'], db_data['insecure'])
+
+
+
+@click.group(cls=AbbreviationGroup)
+def kubernetes():
+    """kubernetes command line"""
+    pass
+
+config.add_command(kubernetes)
```

Comment:
If you move this line into main.py, you don't need to bypass LGTM alert.

Reply:
No. That brings in another alert, because each file is importing each other. BTW, that was my  original code.

Thanks for the suggestion, though.

Reply:
True, but then we get a cyclical import alert. Renuka and I discussed this in the comments above. We have other files causing the cyclical import alerts. I have opened an issue to track the proper fix: https://github.com/Azure/sonic-utilities/issues/967

Reply:
And you don't need to `from config.main import config`

---
In reply to: [451872241](https://github.com/Azure/sonic-utilities/pull/962#discussion_r451872241) [](ancestors = 451872241)

Reply:
I need other functions from main.py, unless I duplicate.

Reply:
Right solution is re-factoring as suggested by Joe. There are already other issues due to this lack of re-factoring, but suppressed. Need a proper cleanup.